src/cc/CCupgrades.cpp

```diff
@@ -0,0 +1,77 @@
+/******************************************************************************
+ * Copyright © 2021 The SuperNET Developers.                             *
+ *                                                                            *
+ * See the AUTHORS, DEVELOPER-AGREEMENT and LICENSE files at                  *
+ * the top-level directory of this distribution for the individual copyright  *
+ * holder information and the developer policies on copyright and licensing.  *
+ *                                                                            *
+ * Unless otherwise agreed in a custom licensing agreement, no part of the    *
+ * SuperNET software, including this file may be copied, modified, propagated *
+ * or distributed except according to the terms contained in the LICENSE file *
+ *                                                                            *
+ * Removal or modification of this copyright notice is prohibited.            *
+ *                                                                            *
+ ******************************************************************************/
+
+#include <iostream>
+#include "CCupgrades.h"
+
+namespace CCUpgrades {
+
+    class CUpgradesContainer {
+    public:
+        CUpgradesContainer()  {
+
+            // default upgrades: always enable all fixes
+            defaultUpgrades.IsAllEnabled = true;
+
+            // TOKEL
+            ChainUpgrades tokel;
+            tokel.setActivationHeight(CCASSETS_OPDROP_VALIDATE_FIX, CCASSETS_OPDROP_FIX_TOKEL_HEIGHT, UPGRADE_ACTIVE);
+            mChainUpgrades["TOKEL"] = tokel;
+
+            // TKLTEST
+            ChainUpgrades tkltest;
+            tkltest.setActivationHeight(CCASSETS_OPDROP_VALIDATE_FIX, CCASSETS_OPDROP_FIX_TKLTEST_HEIGHT, UPGRADE_ACTIVE);
+            mChainUpgrades["TKLTEST"] = tokel;
```

Comment:
```suggestion
            mChainUpgrades["TKLTEST"] = tkltest;
```
Otherwise, TKLTEST chain upgrades will contain TOKEN chain values. Small example:
```
#include <iostream>
#include "CCupgrades.h"

int main() {
    std::cout << "begin" << std::endl;

    CCUpgrades::SelectUpgrades("TKLTEST");
    const CCUpgrades::ChainUpgrades &cu2 = CCUpgrades::GetUpgrades();
    std::cout << "IsAllEnabled: " << cu2.IsAllEnabled << std::endl;
    
    for (const auto& kv : cu2.mUpgrades) {
        std::cout << "UPGRADE_ID = " << kv.first << ", UpgradeInfo.nActivationHeight = " << kv.second.nActivationHeight 
                  << ", UpgradeInfo.status = " << kv.second.status << std::endl;
    }

    std::cout << CCUpgrades::IsUpgradeActive(158960, cu2, CCUpgrades::UPGRADE_ID::CCASSETS_OPDROP_VALIDATE_FIX) << std::endl;
    std::cout << CCUpgrades::IsUpgradeActive(158961, cu2, CCUpgrades::UPGRADE_ID::CCASSETS_OPDROP_VALIDATE_FIX) << std::endl;
    std::cout << CCUpgrades::IsUpgradeActive(158962, cu2, CCUpgrades::UPGRADE_ID::CCASSETS_OPDROP_VALIDATE_FIX) << std::endl;

    std::cout << CCUpgrades::IsUpgradeActive(243158, cu2, CCUpgrades::UPGRADE_ID::CCASSETS_OPDROP_VALIDATE_FIX) << std::endl;
    std::cout << CCUpgrades::IsUpgradeActive(243159, cu2, CCUpgrades::UPGRADE_ID::CCASSETS_OPDROP_VALIDATE_FIX) << std::endl;
    std::cout << CCUpgrades::IsUpgradeActive(243160, cu2, CCUpgrades::UPGRADE_ID::CCASSETS_OPDROP_VALIDATE_FIX) << std::endl;

    std::cout << "end" << std::endl;
    return 0;
}
```

Output will be:
```
begin
IsAllEnabled: 0
UPGRADE_ID = 1, UpgradeInfo.nActivationHeight = 243159, UpgradeInfo.status = 1
0
0
0
0
1
1
end
```
But expected other.

Reply:
Excellent catch, thank you!
TKLTEST nodes were upgraded yesterday so seems it is better to have it with the tokel settings (but I'll fix the var to have the code clean)

Reply:
Also, about `CCUpgrades` interface. Perhaps this is a rare or impossible case, but what if somebody will try to use `SelectUpgrades` from different threads, like:

**thread1:**
```
CCUpgrades::SelectUpgrades("CHAIN1");
const CCUpgrades::ChainUpgrades &cu1 = CCUpgrades::GetUpgrades();
// do something with cu1
```
**thread2:**
```
CCUpgrades::SelectUpgrades("CHAIN2");
const CCUpgrades::ChainUpgrades &cu2 = CCUpgrades::GetUpgrades();
// do something with cu2
```
There will no guarantees that `GetUpgrades()` will return upgrades for correct chain. I mean, what if actual sequence of executions will be:
```
1. CCUpgrades::SelectUpgrades("CHAIN1"); // pSelectedUpgrades will point in CHAIN1 upgrades
2. CCUpgrades::SelectUpgrades("CHAIN2"); // pSelectedUpgrades will point in CHAIN2 upgrades
3. const CCUpgrades::ChainUpgrades &cu1 = CCUpgrades::GetUpgrades(); // as a result, cu1 will contain CHAIN2 upgrades, instead of expected CHAIN1
```
Of course, personally you will not use `SelectUpgrades` that way, but may be `SelectUpgrades` and `GetUpgrades` should be combined into one function and shouldn't use intermediate variable `pSelectedUpgrades`?

Just thinking out of loud.

Reply:
I made this CCUpgrades class similar to the existing NetworkUpgrades.
SelectUpgrades is supposed to call only once at komodod init for the current chain. It is a small optimisation for not comparing the chain name each time when an upgrade is activated in the code. So calls are always readonly except the call on init